Engine/DataFeeds/Subscription.cs

```diff
@@ -91,37 +100,76 @@ public class Subscription : IEnumerator<SubscriptionData>
+            Security = subscriptionRequest.Security;
             _enumerator = enumerator;
-            IsUniverseSelectionSubscription = isUniverseSelectionSubscription;
-            Configuration = configuration;
+            IsUniverseSelectionSubscription = subscriptionRequest.IsUniverseSubscription;
+            Configuration = subscriptionRequest.Configuration;
             OffsetProvider = timeZoneOffsetProvider;
 
-            UtcStartTime = utcStartTime;
-            UtcEndTime = utcEndTime;
+            UtcStartTime = subscriptionRequest.StartTimeUtc;
+            UtcEndTime = subscriptionRequest.EndTimeUtc;
             RemovedFromUniverse = Ref.CreateReadOnly(() => _removedFromUniverse);
         }
 
+        /// <summary>
+        /// Adds a <see cref="SubscriptionRequest"/> for this subscription
+        /// </summary>
+        /// <param name="subscriptionRequest">The <see cref="SubscriptionRequest"/> to add</param>
+        public bool AddSubscriptionRequest(SubscriptionRequest subscriptionRequest)
+        {
+            if (IsUniverseSelectionSubscription
+                || subscriptionRequest.IsUniverseSubscription)
+            {
+                throw new Exception("Subscription.AddSubscriptionRequest(): Universe selection" +
+                    " subscriptions should not have more than 1 SubscriptionRequest");
+            }
+
+            // this shouldn't happen but just in case..
+            if (subscriptionRequest.Configuration != Configuration)
+            {
+                throw new Exception("Subscription.AddSubscriptionRequest(): Requesting to add" +
+                    "a different SubscriptionDataConfig");
+            }
+
+            // Only allow one subscription request per universe
+            if (!Universe.Contains(subscriptionRequest.Universe))
+            {
+                _subscriptionRequests.Add(subscriptionRequest);
+                // TODO this might update the 'UtcStartTime' and 'UtcEndTime' of this subscription
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Removes one or all <see cref="SubscriptionRequest"/> from this subscription
+        /// </summary>
+        /// <param name="universe">Universe requesting to remove <see cref="SubscriptionRequest"/>.
+        /// Default value, null, will remove all universes</param>
+        public IEnumerable<Universe> RemoveSubscriptionRequest(Universe universe = null)
+        {
+            if (universe == null)
+            {
+                var result = _subscriptionRequests;
+                _subscriptionRequests = new List<SubscriptionRequest>();
+                return result.Where(x => x.Universe != null)
+                    .Select(x => x.Universe);
+            }
+
+            _subscriptionRequests.RemoveAll(x => x.Universe == universe);
+            // TODO this might update the 'UtcStartTime' and 'UtcEndTime' of this subscription
```

Comment:
Luckily start is always now and end is always end of algo, although support for dynamic end values may be useful in the future, so definitely good to be considering this.